services/openvpn/service/manager.go

```diff
@@ -236,27 +234,36 @@ func (m *Manager) ProvideConfig(_ string, sessionConfig json.RawMessage) (*sessi
 			return nil, errors.Wrap(err, "could not parse consumer config")
 		}
 
-		if m.location.BehindNAT() && m.portMappingFailed() {
-			pp, err := m.natPingerPorts.Acquire()
+		if len(consumerConfig.Ports) == 0 {
+			cp, err := m.natPingerPorts.Acquire()
 			if err != nil {
 				return nil, err
 			}
 
-			cp, err := m.natPingerPorts.Acquire()
-			if err != nil {
-				return nil, err
+			consumerConfig.Ports = []int{cp.Num(), cp.Num(), cp.Num(), cp.Num()}
+			vpnConfig.LocalPort = cp.Num()
+		}
+
+		if m.location.BehindNAT() && m.portMappingFailed() {
+			for range consumerConfig.Ports {
+				pp, err := m.natPingerPorts.Acquire()
+				if err != nil {
+					return nil, err
+				}
+
+				vpnConfig.Ports = append(vpnConfig.Ports, pp.Num())
+				vpnConfig.RemotePort = pp.Num()
```

Comment:
Maybe `vpnConfig.RemotePort` can be assign once outside of this loop if this is only a single port?

Reply:
This is part of the backward compatibility and it will be removed soon. 
Now RemotePort should be assigned to the latest version. 
I'd prefer to assign it 10 times with a single line, instead of adding extra block with checking slices length and assigning it outside.